### PR TITLE
SNOW-170115 ODBC put on windows fails when file paths have forward slash

### PR DESCRIPTION
**Pull request by sfc-gh-vreddy**
On windows file path C:/Users/vreddy/TestData/testput.csv fails to upload.

This fix will address this issue.

### CODE REVIEW

**Comment by sfc-gh-stakeda** [comment]
you may want to add test cases?

**Comment by sfc-gh-vreddy** [comment]
@smtakeda Added test cases for different file types and URIs in snowflake-odbc

**Comment by sfc-gh-kwagner** [comment]
@sfc-gh-vreddy can you add a unit test for this change in libsnowflakeclient in this file: https://github.com/snowflakedb/libsnowflakeclient/blob/master/tests/test_unit_file_metadata_init.cpp

**Comment by sfc-gh-stakeda** [comment]
@sfc-gh-vreddy ,the best practice is to have test cases next to the the code so that we can catch any error or behavior change as soon as we make the code change. If we have tests in ODBC, then it will be delayed to the stage. Please add the test cases to libsnowflakeclient. Thanks.

File: tests/test_unit_file_metadata_init.cpp [hunk 1]
```diff
@@ -28,33 +28,75 @@ std::string getTestFileMatchDir()
...
+std::vector<std::string> getListOfTestFileMatchDir()
+{
+  std::vector<std::string> listOfTestDirs; 
+  #ifdef _WIN32
+  std::string srcLocation = getTestFileMatchDir();
+  replaceStrAll(srcLocation,"\\","/");
+  // Directory path looks like C:/Users/vreddy/App/Temp/adkf-kasdfk-adsfl/
+  // Testing forward slash for put/get support
+  listOfTestDirs.push_back(srcLocation);
+  #endif
```
Comment (sfc-gh-stakeda) [inline tests/test_unit_file_metadata_init.cpp:63]: Looks like you are deleting the test case including backslashes? Don't we want to test both cases on Windows?

Comment (sfc-gh-vreddy) [reply tests/test_unit_file_metadata_init.cpp:63]: @smtakeda 
The above code pushes new directory to the vector of directories. And line 64,65 is the old directory pushed to vector. We did not delete we are adding all the directories to the vector and running tests on all. 
Line 74 goes through all directories and runs tests.

**Pull request merged**